# Employee Exit Survey Analysis

## Introduction

This project analyzes exit surveys from former employees of two educational institutions in Queensland, Australia: the Department of Education, Training and Employment (DETE) and the Technical and Further Education (TAFE) institute. The goal is to uncover insights into employee resignations and potential dissatisfaction factors.

Our main objectives are to investigate:

1. Whether short-term employees are resigning due to dissatisfaction compared to long-term employees.
2. If younger employees are more likely to resign due to dissatisfaction than older employees.

We'll be working with two datasets:
- DETE exit survey data
- TAFE exit survey data 

This analysis will combine results from both surveys to provide comprehensive insights for our stakeholders. Throughout this notebook, we'll focus on data cleaning, preprocessing, and initial analysis of the first research question.

Let's begin by importing the necessary libraries and loading our datasets into pandas for exploration and analysis.

In [1]:
import pandas as pd
import numpy as np

dete_survey = pd.read_csv("Dataset/05_dete_survey.csv")
tafe_survey = pd.read_csv("Dataset/05_tafe_survey.csv")


### Discovery of the dataset : dete_survey

In [40]:
dete_survey.head()

ID                    SeparationType Cease Date  DETE Start Date  \
0   1             Ill Health Retirement    08/2012           1984.0   
1   2  Voluntary Early Retirement (VER)    08/2012              NaN   
2   3  Voluntary Early Retirement (VER)    05/2012           2011.0   
3   4         Resignation-Other reasons    05/2012           2005.0   
4   5                    Age Retirement    05/2012           1970.0   

   Role Start Date                                      Position  \
0           2004.0                                Public Servant   
1              NaN                                Public Servant   
2           2011.0                               Schools Officer   
3           2006.0                                       Teacher   
4           1989.0  Head of Curriculum/Head of Special Education   

  Classification              Region                      Business Unit  \
0        A01-A04      Central Office  Corporate Strategy and Peformance   
1        AO5-AO7      Central Office  Corporate Strategy and Peformance   
2            NaN      Central Office               Education Queensland   
3        Primary  Central Queensland                                NaN   
4            NaN          South East                                NaN   

     Employment Status  ...  Kept informed  Wellness programs  \
0  Permanent Full-time  ...              N                  N   
1  Permanent Full-time  ...              N                  N   
2  Permanent Full-time  ...              N                  N   
3  Permanent Full-time  ...              A                  N   
4  Permanent Full-time  ...              N                  A   

   Health & Safety  Gender          Age  Aboriginal  Torres Strait  South Sea  \
0                N    Male        56-60         NaN            NaN        NaN   
1                N    Male        56-60         NaN            NaN        NaN   
2                N    Male  61 or older         NaN            NaN        NaN   
3                A  Female        36-40         NaN            NaN        NaN   
4                M  Female  61 or older         NaN            NaN        NaN   

   Disability  NESB  
0         NaN   Yes  
1         NaN   NaN  
2         NaN   NaN  
3         NaN   NaN  
4         NaN   NaN  

[5 rows x 56 columns]

In [5]:
dete_survey.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 822 entries, 0 to 821
Data columns (total 56 columns):
 #   Column                               Non-Null Count  Dtype 
---  ------                               --------------  ----- 
 0   ID                                   822 non-null    int64 
 1   SeparationType                       822 non-null    object
 2   Cease Date                           822 non-null    object
 3   DETE Start Date                      822 non-null    object
 4   Role Start Date                      822 non-null    object
 5   Position                             817 non-null    object
 6   Classification                       455 non-null    object
 7   Region                               822 non-null    object
 8   Business Unit                        126 non-null    object
 9   Employment Status                    817 non-null    object
 10  Career move to public sector         822 non-null    bool  
 11  Career move to private sector        822 non-

In [6]:
dete_survey.isnull().sum()

ID                                       0
SeparationType                           0
Cease Date                               0
DETE Start Date                          0
Role Start Date                          0
Position                                 5
Classification                         367
Region                                   0
Business Unit                          696
Employment Status                        5
Career move to public sector             0
Career move to private sector            0
Interpersonal conflicts                  0
Job dissatisfaction                      0
Dissatisfaction with the department      0
Physical work environment                0
Lack of recognition                      0
Lack of job security                     0
Work location                            0
Employment conditions                    0
Maternity/family                         0
Relocation                               0
Study/Travel                             0
Ill Health 

Percentage of null values by category

In [13]:
null_rate_dete_survey = dete_survey.isnull().sum() / len(dete_survey) * 100

In [14]:
null_rate_dete_survey

ID                                      0.000000
SeparationType                          0.000000
Cease Date                              0.000000
DETE Start Date                         0.000000
Role Start Date                         0.000000
Position                                0.608273
Classification                         44.647202
Region                                  0.000000
Business Unit                          84.671533
Employment Status                       0.608273
Career move to public sector            0.000000
Career move to private sector           0.000000
Interpersonal conflicts                 0.000000
Job dissatisfaction                     0.000000
Dissatisfaction with the department     0.000000
Physical work environment               0.000000
Lack of recognition                     0.000000
Lack of job security                    0.000000
Work location                           0.000000
Employment conditions                   0.000000
Maternity/family    

The dataset dete_survey contains 822 entries with 56 columns, including employee details, separation reasons, and demographic information. While most columns have complete data, some fields like Classification, Business Unit, and certain demographic categories (e.g., Aboriginal, Torres Strait) have significant missing values.

### Discovery of the dataset : tafe_survey

In [10]:
tafe_survey.head()

Record ID                              Institute  \
0  6.341330e+17  Southern Queensland Institute of TAFE   
1  6.341337e+17            Mount Isa Institute of TAFE   
2  6.341388e+17            Mount Isa Institute of TAFE   
3  6.341399e+17            Mount Isa Institute of TAFE   
4  6.341466e+17  Southern Queensland Institute of TAFE   

                   WorkArea  CESSATION YEAR Reason for ceasing employment  \
0  Non-Delivery (corporate)          2010.0              Contract Expired   
1  Non-Delivery (corporate)          2010.0                    Retirement   
2       Delivery (teaching)          2010.0                    Retirement   
3  Non-Delivery (corporate)          2010.0                   Resignation   
4       Delivery (teaching)          2010.0                   Resignation   

  Contributing Factors. Career Move - Public Sector   \
0                                                NaN   
1                                                  -   
2                                                  -   
3                                                  -   
4                                                  -   

  Contributing Factors. Career Move - Private Sector   \
0                                                NaN    
1                                                  -    
2                                                  -    
3                                                  -    
4                       Career Move - Private Sector    

  Contributing Factors. Career Move - Self-employment  \
0                                                NaN    
1                                                  -    
2                                                  -    
3                                                  -    
4                                                  -    

  Contributing Factors. Ill Health Contributing Factors. Maternity/Family  \
0                              NaN                                    NaN   
1                                -                                      -   
2                                -                                      -   
3                                -                                      -   
4                                -                                      -   

   ...  \
0  ...   
1  ...   
2  ...   
3  ...   
4  ...   

  Workplace. Topic:Does your workplace promote a work culture free from all forms of unlawful discrimination?  \
0                                                Yes                                                            
1                                                Yes                                                            
2                                                Yes                                                            
3                                                Yes                                                            
4                                                Yes                                                            

  Workplace. Topic:Does your workplace promote and practice the principles of employment equity?  \
0                                                Yes                                               
1                                                Yes                                               
2                                                Yes                                               
3                                                Yes                                               
4                                                Yes                                               

  Workplace. Topic:Does your workplace value the diversity of its employees?  \
0                                                Yes                           
1                                                Yes                           
2                                                Yes                           
3                                              

In [17]:
tafe_survey.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 702 entries, 0 to 701
Data columns (total 72 columns):
 #   Column                                                                                                                                                         Non-Null Count  Dtype  
---  ------                                                                                                                                                         --------------  -----  
 0   Record ID                                                                                                                                                      702 non-null    float64
 1   Institute                                                                                                                                                      702 non-null    object 
 2   WorkArea                                                                                                                                  

In [16]:
null_rate_tafe_survey = tafe_survey.isnull().sum() / len(tafe_survey) * 100

In [18]:
null_rate_tafe_survey

Record ID                                                                     0.000000
Institute                                                                     0.000000
WorkArea                                                                      0.000000
CESSATION YEAR                                                                0.997151
Reason for ceasing employment                                                 0.142450
                                                                               ...    
CurrentAge. Current Age                                                      15.099715
Employment Type. Employment Type                                             15.099715
Classification. Classification                                               15.099715
LengthofServiceOverall. Overall Length of Service at Institute (in years)    15.099715
LengthofServiceCurrent. Length of Service at current workplace (in years)    15.099715
Length: 72, dtype: float64

The dataset tafe_survey contains 702 entries with 72 columns.

### Identify Missing Values and Drop Unnecessary Columns

In [2]:
dete_survey = pd.read_csv("Dataset/05_dete_survey.csv", na_values = "Not Stated")

In [4]:
dete_survey_updated = dete_survey.drop(dete_survey.columns[28:49], axis=1)
tafe_survey_updated = tafe_survey.drop(tafe_survey.columns[17:66], axis=1)

In our analysis, we identified that the dete_survey DataFrame contained 'Not Stated' values that indicate missing data, but these values were not represented as NaN. To improve the quality of our data and facilitate analysis, we replaced these values with NaN using the pd.read_csv() function with the na_values parameter.

Additionally, we found that both the dete_survey and tafe_survey DataFrames contain many columns that are not necessary for our analysis. To streamline our DataFrames and focus on relevant data, we used the .drop() method to remove unnecessary columns. This will allow us to perform a more efficient and targeted analysis.

### Clean Column Names

In [5]:
dete_survey_updated.columns = dete_survey_updated.columns.str.upper().str.strip().str.replace(" ","_")

In [6]:
dete_survey_updated

ID                        SEPARATIONTYPE CEASE_DATE  DETE_START_DATE  \
0      1                 Ill Health Retirement    08/2012           1984.0   
1      2      Voluntary Early Retirement (VER)    08/2012              NaN   
2      3      Voluntary Early Retirement (VER)    05/2012           2011.0   
3      4             Resignation-Other reasons    05/2012           2005.0   
4      5                        Age Retirement    05/2012           1970.0   
..   ...                                   ...        ...              ...   
817  819                        Age Retirement    02/2014           1977.0   
818  820                        Age Retirement    01/2014           1980.0   
819  821  Resignation-Move overseas/interstate    01/2014           2009.0   
820  822                 Ill Health Retirement    12/2013           2001.0   
821  823  Resignation-Move overseas/interstate    12/2013              NaN   

     ROLE_START_DATE                                      POSITION  \
0             2004.0                                Public Servant   
1                NaN                                Public Servant   
2             2011.0                               Schools Officer   
3             2006.0                                       Teacher   
4             1989.0  Head of Curriculum/Head of Special Education   
..               ...                                           ...   
817           1999.0                                       Teacher   
818           1980.0                                       Teacher   
819           2009.0                                Public Servant   
820           2009.0                                       Teacher   
821              NaN                                  Teacher Aide   

    CLASSIFICATION                    REGION  \
0          A01-A04            Central Office   
1          AO5-AO7            Central Office   
2              NaN            Central Office   
3          Primary        Central Queensland   
4              NaN                South East   
..             ...                       ...   
817        Primary        Central Queensland   
818      Secondary               North Coast   
819        A01-A04            Central Office   
820      Secondary  Darling Downs South West   
821            NaN              Metropolitan   

                         BUSINESS_UNIT    EMPLOYMENT_STATUS  ...  \
0    Corporate Strategy and Peformance  Permanent Full-time  ...   
1    Corporate Strategy and Peformance  Permanent Full-time  ...   
2                 Education Queensland  Permanent Full-time  ...   
3                                  NaN  Permanent Full-time  ...   
4                                  NaN  Permanent Full-time  ...   
..                                 ...                  ...  ...   
817                                NaN  Permanent Part-time  ...   
818                                NaN  Permanent Full-time  ...   
819               Education Queensland  Permanent Full-time  ...   
820                                NaN  Permanent Full-time  ...   
821                                NaN                  NaN  ...   

     WORK_LIFE_BALANCE  WORKLOAD  NONE_OF_THE_ABOVE  GENDER          AGE  \
0                False     False               True    Male        56-60   
1                False     False              False    Male        56-60   
2                False     False               True    Male  61 or older   
3                False     False              False  Female        36-40   
4                 True     False              False  Female  61 or older   
..                 ...       ...                ...     ...          ...   
817              False      True              False  Female        56-60   
818              False     False               True    Male        51-55   
819               True     False              False  Female        31-35   
820              False     False              False  Female        41-45  

In [28]:
tafe_survey_updated = tafe_survey.rename(columns={
    'id': 'Record ID',
    'cease_date': 'CESSATION YEAR',
    'separationtype': 'Reason for ceasing employment',
    'gender': 'Gender. What is your Gender?',
    'age': 'CurrentAge. Current Age',
    'employment_status': 'Employment Type. Employment Type',
    'position': 'Classification. Classification',
    'institute_service': 'LengthofServiceOverall. Overall Length of Service at Institute (in years)',
    'role_service': 'LengthofServiceCurrent. Length of Service at current workplace (in years)'
})